Add configure GPG locations

mamedin opened the pull request:
When the `archivematica_src_configure_gpg` is defined, this commit adds
a new GPG space if it doesn't already exist and configures a GPG AIPs
Store and a GPG Transfer Backlog.

The main purpose of the automatic configuration of GPG space and
locations is for QA and AMAUAT. For production sites it is better to
create GPG Space and locations from the GUI.

Connects to https://github.com/archivematica/Issues/issues/832

Commit:
Add configure GPG locations

When the `archivematica_src_configure_gpg` is defined, this commit adds
a new GPG space if it doesn't already exist and configures a GPG AIPs
Store and a GPG Transfer Backlog.

The main purpose of the automatic configuration of GPG space and
locations is for QA and AMAUAT. For production sites it is better to
create GPG Space and locations from the GUI.

scollazo commented on tasks/configure-gpg.yml:
```diff
@@ -0,0 +1,156 @@
+
+- name: "Count GPG Spaces"
+  shell: > 
+    curl -s -X GET 
```
I would prefer to use always the url ansible module, and set the fact in a different task, using jq or jinja2 filters.

scollazo commented on tasks/configure-gpg.yml:
```diff
@@ -0,0 +1,156 @@
+    - name: "Set gpg binary name"
+      set_fact:
+        gpg_binary: "gpg1"
+      when: "ansible_distribution_version | version_compare('18.04', '>=')"
```
Is this correct? We don't support ubuntu >= 18.04

mamedin replied:
Yes, for future versions :P

scollazo replied:
It's always good to think about the future ;)

scollazo commented on tasks/configure-gpg.yml:
```diff
@@ -0,0 +1,156 @@
+
+    - name: "Check whether GPG already exists"
+      shell: >
+        {{ archivematica_src_configure_gpg.binary }} --no-tty 
```
Could you split each option into it's own line?

scollazo commented on templates/gpg/gpg-batch-file.j2:
```diff
@@ -0,0 +1,13 @@
+Name-Comment: {{ archivematica_src_configure_gpg.key_comment }}
+Name-Email: {{ archivematica_src_configure_gpg.key_mail }}
+Expire-Date: 0
+Passphrase: passphrase
```
Can we use a configuration parameter for this, and if it's not available just use a blank passphrase?

mamedin replied:
We are using empty string when creating the keys from GUI:

https://github.com/artefactual/archivematica-storage-service/blob/stable/0.14.x/storage_service/common/gpgutils.py#L31

We can add a new field to the dict for the passphrase (it's free), good idea. Thanks!

mamedin commented:
Hi Santi:

* removed the jq/curl dependency: Using uri module and json_query filter
* Added passphrase var
* changed shell tasks with 1 line for each option.

scollazo commented on tasks/configure-gpg.yml:
```diff
@@ -0,0 +1,159 @@
+    - name: "Create batch file for GPG key"
+      template:
+        src: "gpg/gpg-batch-file.j2"
+        dest: "/tmp/gpg-batch-file"
```
Is this file deleted after the process finishes?

mamedin replied:
Good catch!!

scollazo reviewed (approved):
Other than removing the temporary gpg batch file, this is ready for merge!
